packages/sources/spectral-macro-score/src/adapter.ts

```diff
@@ -0,0 +1,44 @@
+import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
+import { ExecuteFactory, AdapterRequest, AdapterResponse } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT, SpectralAdapterConfig } from './config'
+import { MacroScoreAPI } from './endpoint'
+
+const inputParams = {
+  tokenIdInt: true,
+  tickSetId: true,
+}
+
+export const execute = async (
+  request: AdapterRequest,
+  config: SpectralAdapterConfig,
+): Promise<AdapterResponse> => {
+  console.log('execute')
+  const validator = new Validator(request, inputParams)
+  console.log('WUT')
+  if (validator.error) {
+    console.log(JSON.stringify(validator.error))
+    throw validator.error
+  }
+
+  Requester.logConfig(config)
+
+  request.data.jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint.toLowerCase()) {
+    case MacroScoreAPI.MacroScoreAPIName: {
+      return await MacroScoreAPI.execute(request, config)
+    }
+    default: {
+      throw new AdapterError({
+        jobRunID: request.data.jobRunID,
+        message: `Endpoint ${endpoint} not supported.`,
+        statusCode: 400,
+      })
+    }
+  }
+}
+
+export const makeExecute: ExecuteFactory<SpectralAdapterConfig> = () => {
+  return async (request) => execute(request, makeConfig())
+}
```

Comment:
Change with context support:

```suggestion
export const makeExecute: ExecuteFactory<SpectralAdapterConfig> = (config) => {
  return async (request, context) => execute(request, context, config || makeConfig())
}
```